Example/navigation/index.js

```diff
@@ -35,57 +30,56 @@ const Background = ({ index }) => (
   />
 );
 
-class DetailsScreen extends React.Component {
+const DetailsScreen = ({ navigation, route }) => {
   animvalue = new Animated.Value(0);
-  rotation = this.animvalue.interpolate({
+  rotation = animvalue.interpolate({
     inputRange: [0, 1],
     outputRange: ['0deg', '360deg'],
   });
-  state = { count: 1, text: '' };
+  const [count, setCount] = useState(1);
+  const [text, setText] = useState('');
 
-  componentDidMount() {
-    this.props.navigation.setOptions({
-      title: `Details screen #${this.getIndex()}`,
+  useEffect(() => {
+    navigation.setOptions({
+      title: `Details screen #${getIndex()}`,
     });
     Animated.loop(
-      Animated.timing(this.animvalue, {
+      Animated.timing(animvalue, {
         toValue: 1,
         duration: 1000,
         useNativeDriver: false,
       })
     ).start();
-    setInterval(() => this.setState({ count: this.state.count + 1 }), 500);
-  }
+    setInterval(() => setCount(count + 1 ), 1000);
+  },[]); 
```

Comment:
Shouldn't `navigation` be included in the dependencies array? Same for other `useEffect`.